models/enterprise_sonic/snmp/sonic_snmp.yml

```diff
@@ -1,53 +1,337 @@
+                    type: str
+                    choices:
+                      - no-auth-no-priv
+                      - auth-no-priv
+                      - auth-priv
+                    required: true
+          communities:
+            description:
+              - List of communities
+            type: list
+            elements: dict
+            suboptions:
+              index:
+                description:
+                  - Index
+                type: str
+                required: true
+              security_name:
+                description:
+                  - Security name
+                type: str
+          engine:
+            description:
+              - SNMP engine configuration
+            type: dict
+            suboptions:
+              enabled:
+                description:
+                type: bool
+              listen:
+                description:
+                  - List of configurations for the transport endpoints on which the engine listens
+                type: list
+                elements: dict
+                suboptions:
+                  name:
+                    description:
+                      - Name of listen entry
+                    type: str
+                    required: true
+                  udp:
+                    description:
+                      - UDP configuration
+                    type: dict
+                    suboptions:
+                      ip:
+                        description:
+                          - IP address on which the engine listens
+                        type: str
+                      port:
+                        description:
+                          -  UDP port number on which the engine listens, range 1024-65535
+                        type: int
+                      interface:
+                        description:
+                          - Name of interface
+                        type: str
+              engine_id:
+                description:
+                  - Engine's administratively assigned unique identifier
+                type: str
+              version:
+                description:
+                type: dict
+                options:
+                  v2c:
+                    description:
+                      - If v2c is enabled
+                    type: bool
+                  v3:
+                    description:
+                      - If v3 is enabled
+                    type: bool
+          users:
+            description:
+              - List of users
+            type: list
+            elements: dict
+            suboption:
+              name:
+                description:
+                  - Name of user
+                type: str
+                required: true
+              auth:
+                description:
+                  - User authentication configuration
+                type: dict
+                suboptions:
+                  md5_key: 
```

Comment:
Are md5 and sha authentication modes mutually exclusive? If so, they need to be specified that way in the model.

Reply:
I did some investigation of this and they are mutually exclusive.

Please add an "auth_type" suboption under "auth" that can be set to either "md5" or "sha" (using a "choices" attribute for the "auth_type" option). Then, provide one "key" option to be used with whichever auth_type was selected (instead of having both an md5_key and sha_key option).